exporter/splunkhecexporter/client.go

```diff
@@ -148,19 +152,150 @@ func (c *client) sendSplunkEvents(ctx context.Context, splunkEvents []*splunk.Ev
 	return nil
 }
 
-func (c *client) pushLogData(ctx context.Context, ld pdata.Logs) error {
+func subLogs(ld *pdata.Logs, start *logIndex) *pdata.Logs {
+	if ld == nil || start == nil || start.zero() {
+		return ld
+	}
+
+	logs := pdata.NewLogs()
+	RL, RL2 := ld.ResourceLogs(), logs.ResourceLogs()
+
+	for r := start.resource; r < RL.Len(); r++ {
+		RL2.Append(pdata.NewResourceLogs())
```

Comment:
Could possibly reduce allocations by doing RL2.Resize(X) then using RL2.At() instead of Append. Same for some others below. Can do in subsequent PR if needed.